src/core/common/Sidebar/functions.js

```diff
@@ -0,0 +1,46 @@
+import { get } from "lodash";
+import { resourceOrder } from "../../../version/config/theme.config";
+
+export const SHORT_MENU = "short";
+export const FULL_MENU = "full";
+
+/**
+ * This function sorts resources by theme settings
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ * @param {array} resources
+ * @return {array}
+ */
+export function sortResourcesArray(resources) {
+    let sortResource = [];
+    for (let i = 0, n = resourceOrder.length; i < n; i++) {
+        let currentItem = resourceOrder[i];
+        for (let j = 0, m = resources.length; j < m; j++) {
+            if (currentItem === get(resources, [j, 'name'], null)) {
+                sortResource[i] = resources[j];
+                break;
+            }
+        }
+    }
+    return sortResource;
+}
+
+/**
+ * This function returns type of menu
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ * @param {string} currentPathname
+ * @return {string}
+ */
+export function getMenuType(currentPathname) {
```

Comment:
Your screencast doesn't show the URL bar, but I assume that when the app displays patient details pages, they are all under the same parent url (something like `/patientDetails/allergies`, or `/patient/1234/allergies`). So the URL carries enough state to let you decide which menu to display: if it starts with `/patientDetails`, then display the long menu, otherwise display the short menu.

React-router's `<Route>` component triggers the rendering of a component depending on the URL. This is exactly what you need here, combined with a `<Switch>`.

https://reacttraining.com/react-router/web/api/Route